pkg/networkservice/common/monitor/server.go

```diff
@@ -79,14 +79,20 @@ func (m *monitorServer) MonitorConnections(selector *networkservice.MonitorScope
 
 func (m *monitorServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
 	conn, err := next.Server(ctx).Request(ctx, request)
-	eventConn := conn.Clone()
 	if err == nil {
+		if len(conn.GetPath().PathSegments) == 0 {
+			return nil, errors.New("resp.Path.PathSegments has inccorect lengrh = 0")
+		}
+		eventConn := conn.Clone()
 		m.executor.AsyncExec(func() {
-			m.connections[eventConn.GetId()] = eventConn
```

Comment:
The root cause here: nsmgr could duplicate request from nsc and from forwarder here(because they have different connection id).